### PR TITLE
update sigcpp from 2.6.2 to 3.0.7

### LINKED ISSUE #7440: Update libsigc++ to latest 3.0.X version
libsigc++ sources are now available via github https://github.com/libsigcplusplus/libsigcplusplus/tree/3.0.7 . Newer version might fix the issues with ASAN IB tests ( see https://github.com/cms-sw/cmssw/issues/36056 ).

@ddaina , can you please take care of it?

### PR DESCRIPTION
**Pull request by ddaina**
resolves  #7440

### CODE REVIEW

**Comment by smuzaffar** [comment]
thanks @ddaina , can you please use the sources from https://github.com/libsigcplusplus/libsigcplusplus/archive/refs/tags/%{realversion}.tar.gz and update https://github.com/cms-sw/cmsdist/pull/7441/files#diff-a74d26461bdcbb7dcdd15733528160d42c03aee4744b56b24d6e6e92546e0303R8 is needed?

**Comment by smuzaffar** [comment]
@ddaina , please check if https://github.com/cms-sw/cmsdist/pull/7441/files#diff-a74d26461bdcbb7dcdd15733528160d42c03aee4744b56b24d6e6e92546e0303L26 is correct? with sigccp version 3 we might have to fix this too? I would suggest to try to build locally first.

**Comment by smuzaffar** [comment]
please test

File: sigcpp.spec [hunk 1]
```diff
@@ -1,25 +1,29 @@
 ### RPM external sigcpp 3.0.7
 Source: https://github.com/libsigcplusplus/libsigcplusplus/archive/refs/tags/%{realversion}.tar.gz
 
-BuildRequires: autotools
+BuildRequires: cmake
```
Comment (smuzaffar) [inline sigcpp.spec]: add `gmake` dependency too

Comment (ddaina) [reply sigcpp.spec]: thanks, done

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
FYI @alja , looks like fireworks needs changes for sigc++ 3.0 ( I think this is due to the change https://github.com/libsigcplusplus/libsigcplusplus/blob/master/NEWS#L275 )

**Comment by alja** [comment]
@smuzaffar I see from your link that the syntax of the declaration has changed. The syntax is now like in std::function. In the error from https://github.com/cms-sw/cmssw/blob/master/Fireworks/Core/interface/CSGAction.h#L100 the declaration I think should be sigc::signal<void()>

Should I do custom cms build to test FWLite with new sigc?

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@alja , no need to rebuild fwlite, just a cmssw PR with fixes should be enough. In order to make cmssw changes, please create a local cmssw dev area and run `scram setup /cvmfs/cms-ci.cern.ch/week0/PR_0974cfab/slc7_amd64_gcc900/cms/cmssw-tool-conf/52.0-4f7adebb60be374180e6afb6c6730415/tools/selected/sigcpp.xml` and checkout all the FWLite packages and update the sigc++ interface. Once cmssw PR is available then we can test that along with this cmsdist PR

**Comment by alja** [comment]
@smuzaffar 
I have made changes for the transition to sigcpp3 in commit https://github.com/alja/cmssw/commit/9f144af638866361f7961b16d3dd2483b6974d0c, in branch https://github.com/alja/cmssw/tree/sigcpp3 split from from-CMSSW_12_2_X_2021-11-21-2300

I have made unit test and GUI tests that was related.

Let me know if there is anything else I can do.

**Comment by smuzaffar** [comment]
test parameters:
- pull_request = cms-sw/cmssw#36343

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
thanks @alja , I have opened a cms-sw/cmssw PR cms-sw/cmssw#36343 using your branch for sigcpp 3 changes

**Comment by smuzaffar** [comment]
hold
we will integrate in in 12.3.X

**Comment by smuzaffar** [comment]
please test for slc7_ppc64le_gcc9

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc10

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc11

**Comment by smuzaffar** [comment]
unhold

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals

**Comment by perrotta** [comment]
+1
- It needs https://github.com/cms-sw/cmssw/pull/36343

**Pull request merged**